### DISCUSSION TITLE
Max Workflows on K8s Cluster / Argo Workflow Limits

### DISCUSSION THREAD

**Discussion by TekTimmy** [argoproj/argo-workflows#5403] [Q&A]
I stumpled across these Kubernetes limit information:
> A cluster is a set of nodes (physical or virtual machines) running Kubernetes agents, managed by the control plane. Kubernetes v1.20 supports clusters with up to 5000 nodes. More specifically, Kubernetes is designed to accommodate configurations that meet all of the following criteria:
    * No more than 100 pods per node
    * No more than 5000 nodes
    * No more than 150000 total pods
    * No more than 300000 total containers

Source: https://kubernetes.io/docs/setup/best-practices/cluster-large/

When I get it right this means that if I have 10,000 workflows with each 15 workflow steps Argo creates 150,000 pods and 300,000 containers (main & watch container) which is already the max number Kubernetes can handle...
I know that those pods / container are not running in parallel but that's also not what those limits are about (as far as I understand).

Am I getting it right? What is your experience?

**Comment by jitendra-1217** [comment]
@TekTimmy : Please help if you got any more info /closer on this since started this discussion

**Comment by alexec** [comment]
This is correct, but in practice we see users encounter other practical limits before they reach these theoretical limits. We've seen users run 1000s of workflows a day, and the largest workflows have over 25k nodes. The only way you can tell if you cluster can run  massive workflows is to try.
